pkg/provisioner.go

```diff
@@ -50,113 +51,94 @@ func (s *ProvisionerServer) ProvisionerCreateBucket(ctx context.Context,
 		klog.ErrorS(errors.New("Invalid Argument"), "S3 protocol is nil")
 		return nil, status.Error(codes.InvalidArgument, "S3 Protocol is nil")
 	}
-
-	bucketName := s3.BucketName
+	//TODO : validate S3 protocol defined, check points valid rgwendpoint, v4 signature check etc
+	bucketName := req.GetName()
 	klog.V(3).InfoS("Create Bucket", "name", bucketName)
 
-	err := s.S3Client.CreateBucket(s3.BucketName)
+	err := s.S3Client.CreateBucket(bucketName)
 	if err != nil {
 		// Check to see if the bucket already exists by above api
```

Comment:
Is this TODO ?

Reply:
Yes

Reply:
@jeffvance I am also a bit confused, do I need to create different users for each `bucketRequest` instead of the admin user passed to the COSI driver during init()? 
current workflow:
Admin user -> creates bucket on BucketRequest.
Or different approach
Admin User -> creates a user -> creates bucket on BucketRequest.

Admin user details will be exposed as secret in the namespace, so if the user has that access. Then he can any buckets available